pkg/config/config.go

```diff
@@ -157,32 +170,60 @@ func dirExists(path string) bool {
 // BindFlags binds each cobra flag to its associated viper configuration (config file and environment variable),
 // and validate required flag(s)
 func BindFlags(cmd *cobra.Command, configPrefix string, set map[string]bool) (err error) {
-	keys := make([]string, 0)
 	cmd.Flags().VisitAll(func(f *pflag.Flag) {
-		// Apply the viper config value to the flag when the flag is not set and viper has a value
 		if !f.Changed {
-			if viper.IsSet(configPrefix + f.Name) {
-				val := viper.Get(configPrefix + f.Name)
-				err = cmd.Flags().Set(f.Name, fmt.Sprintf("%v", val))
-				if err != nil {
-					return
-				}
-			} else {
-				// Validate required flags
-				if set[f.Name] {
-					set[f.Name] = false
-					keys = append(keys, f.Name)
-				}
-			}
+			err = setFlagFromConfig(cmd.Flags(), f, configPrefix, set)
 		}
 	})
 	if err != nil {
 		return err
 	}
-	// Return error on required flag(s)
+	return validateRequiredFlags(set)
+}
+
+// initCommonConfig initializes common flags in config file
+// TODO: add common flags for eventing commands
+func initCommonConfig() map[string]map[string]bool {
+	common := make(map[string]map[string]bool)
+	common["service"] = defaultServiceCommonFlags
+	return common
+}
+
+// setFlagsByConfig applies viper config value to flag when flag is not set in command and config has a value
```

Comment:
typo: `setFlagsByConfig` should be `setFlagFromConfig`.

Reply:
addressed